lib/agilecrm-wrapper/deal.rb

```diff
@@ -9,7 +9,22 @@ class Deal < Hashie::Mash
     class << self
 
       def all
-        response = AgileCRMWrapper.connection.get('opportunity')
+        results = []
+        @cursor = nil
+
+        until @no_more_results do
+          response = call_agile_api(@cursor)
+          results.concat(response)
+          @cursor = response.last["cursor"]
+          @no_more_results = @cursor.nil?
+        end
+
+        return results
+      end
+
+      def call_agile_api(cursor = nil)
+        response = AgileCRMWrapper.connection.get("opportunity?cursor=#{cursor}")
+
```

Comment:
not necessary now but for later:

Seeing some close similarities in `deal.rb` & `contact.rb` with main difference mainly the api endpoint. You could make your own little DSL (wrapper class) around `AgileCRMWrapper` where you'd have a class with a method you feed the endpoint. For example: `opportunity?cursor=#{cursor}`. 

Then, wrapped around that is the ` until @no_more_results ..` etc. Future food for thoughts.